### DISCUSSION TITLE
How to implement mobile forgot password screen?

### DISCUSSION THREAD

**Discussion by matart15** [supabase/supabase#20395] [Contribute to Supabase]
I created screen to imput user's email.
As far as I found There is only one function to reset user's password. which is [resetPasswordForEmail](https://supabase.com/docs/reference/javascript/auth-resetpasswordforemail).

So I called it and it sent email to me. 

I used deep linking to redirect back to my mobile app and open `NewPassword` screen.

but it gives me url which returns `{"code": "12dd191b-cac0-4fa1-9bd1-3072
b3c6e0f7"}`

But there is no info on how to use this `code`. 

For web app it seems like it sets session or something. then app can call `supabase.auth.update`, because now `supabase.auth` have authenticated user. 
But for mobile `supabase.auth` have no authenticated user.

**Comment by TiagoLira** [comment]
On iOS I'm using that URL to start a session using the supabase library:
`client.auth.session(from: url)`
Other libraries should have a similar method.
That starts a new session and then I show a custom screen to set a new password.

Hope it helps.

**Comment by knielsen24** [comment]
You are using PKCE flowtype. In the link below look over the overview section.  
https://supabase.com/docs/guides/auth/passwords#overview

This gave me some trouble, and had to piece information together from many places.
